tests/Fixer/Operator/AssignNullCoalescingToCoalesceEqualFixerTest.php

```diff
@@ -0,0 +1,176 @@
+        ];
+
+        yield 'mixed array' => [
+            '<?php
+                $a[1] ??= 1;
+                $a{2} ??= 1;
+                $a{2}[$f] ??= 1;
+            ',
+            '<?php
+                $a[1] = $a[1] ?? 1;
+                $a{2} = $a{2} ?? 1;
+                $a{2}[$f] = $a{2}[$f] ?? 1;
+            ',
+        ];
+
+        yield [
+            '<?php \A\B::$foo ??= 1;',
+            '<?php \A\B::$foo = \A\B::$foo ?? 1;',
+        ];
+
+        yield 'same' => [
+            '<?php $a ??= 1;',
+            '<?php $a = ($a) ?? 1;',
+        ];
+
+        yield 'same II' => [
+            '<?php $a[1] ??= 1;',
+            '<?php $a[1] = $a{1} ?? 1;',
+        ];
+
+        yield 'same III' => [
+            '<?php $a[1] ??= 1;',
+            '<?php $a[1] = (($a{1})) ?? 1;',
+        ];
+
+        yield 'object' => [
+            '<?php $a->b ??= 1;',
+            '<?php $a->b = $a->b ?? 1;',
+        ];
+
+        yield 'object II' => [
+            '<?php $a->b[0]->{1} ??= 1;',
+            '<?php $a->b[0]->{1} = $a->b[0]->{1} ?? 1;',
+        ];
+
+        yield 'simple, before ;' => [
+            '<?php ; $a ??= 1;',
+            '<?php ; $a = $a ?? 1;',
+        ];
+
+        yield 'simple, before {' => [
+            '<?php { $a ??= 1; }',
+            '<?php { $a = $a ?? 1; }',
+        ];
+
+        yield 'simple, before }' => [
+            '<?php if ($a){} $a ??= 1;',
+            '<?php if ($a){} $a = $a ?? 1;',
+        ];
+
+        yield 'simple, before (' => [
+            '<?php foo( $a ??= 1 );',
+            '<?php foo( $a = $a ?? 1 );',
+        ];
+
+        yield 'simple, before ) I' => [
+            '<?php if ($a) $a ??= 1;',
+            '<?php if ($a) $a = $a ?? 1;',
+        ];
+
+        yield 'simple, before ) II' => [
+            '<?php
+                if ($a) $a ??= 1;
+                foreach ($d as $i) $a ??= 1;
+                while (foo()) $a ??= 1;
+            ',
+            '<?php
+                if ($a) $a = $a ?? 1;
+                foreach ($d as $i) $a = $a ?? 1;
+                while (foo()) $a = $a ?? 1;
+            ',
+        ];
+
+        yield 'simple, end' => [
+            '<?php $a ??= 1 ?>',
+            '<?php $a = $a ?? 1 ?>',
+        ];
+
+        yield 'simple, 10x' => [
+            '<?php'.str_repeat(' $a ??= 1;', 10),
+            '<?php'.str_repeat(' $a = $a ?? 1;', 10),
+        ];
+
+        yield 'simple, multi line' => [
+            '<?php
+            $a
+             ??=
+              '.'
+               '.'
+                1;',
+            '<?php
+            $a
+             =
+              $a
+               ??
+                1;',
+        ];
+
+        yield 'do not fix' => [
+            '<?php
+                $a = 1 + $a ?? $b;
+                $b + $a = $a ?? 1;
+                $b = $a ?? 1;
+                $b = $a ?? $b;
+                $d = $a + $c ; $c ?? $c;
```

Comment:
please add these cases to the `do not fix` area:

```php
$a = ($a ?? $b) && $c; // just to be sure
$a = $a ?? $b ? $c : $d; // I think this is not handled correctly at the moment?
```

For the second case: https://3v4l.org/UYuqU